diagnosis/getkuberneteslogs.sh

```diff
@@ -23,11 +22,122 @@ function restore_ssh_config
     fi
 }
 
+login_azs() {
+    local spn_id=$1
+    local spn_secret=$2
+    local tenant_id=$3
+    
+    az login --service-principal -u $spn_id -p $spn_secret --tenant $tenant_id
+}
+
+askSubscription() {
+    az account list -otable
+    echo ""
+    echo "Please enter the Id of the account you wish to use. If you do not see"
+    echo "a valid account in the list press Ctrl+C to abort and create one."
+    echo "If you leave this blank we will use the Current account."
+    echo -n "> "
+    read azure_subscription_id
+
+    if [ "$azure_subscription_id" != "" ]; then
+        az account set --subscription $azure_subscription_id
+    else
+        azure_subscription_id=$(az account list --output json | jq -r '.[] | select(.isDefault==true) | .id')
```

Comment:
This is probably not needed. I would try to remove it as it adds the `jq` dependency